backend/src/services/course-category/course-category.service.ts

```diff
@@ -27,6 +27,18 @@ class CourseCategory {
     };
   }
 
+  public async getWithCourses(): Promise<CategoryGetAllResponseDto> {
+    const categories = await this.#courseCategoryRepository.getWithCourses();
+
+    return {
+      items: categories.map((category) => ({
+        id: category.id,
+        key: category.key,
+        name: category.name,
+      })),
```

Comment:
Is it necessary to map results from `getWithCourses`, since you've already selected only this props in query?